Python/symtable.c

```diff
@@ -543,14 +544,22 @@ analyze_name(PySTEntryObject *ste, PyObject *scopes, PyObject *name, long flags,
                          "nonlocal declaration not allowed at module level");
             return error_at_directive(ste, name);
```

Comment:
Remark unrelated to you PR. I don't understand why an exception is raised before calling error_at_directive() which seems to always override the current exception (and not chain the new one to the currently raised one).

Reply:
No, it only sets location in the raised SyntaxError.

Reply:
I looked again at PyErr_SyntaxLocationObjectEx(): oh, you're right. Now it makes sense, thanks :-)